esp-hal/src/spi/master.rs

```diff
@@ -100,6 +96,43 @@ pub enum SpiInterrupt {
     /// This interrupt is triggered when an SPI transaction has finished
     /// transmitting and receiving data.
     TransDone,
+
+    /// Triggered when the length of GDMA RX FIFO is shorter than that of actual
+    /// data transferred.
+    #[cfg(gdma)]
+    DmaInfifoFullErr,
+
+    /// Triggered when the length of GDMA TX FIFO is shorter than that of actual
+    /// data transferred.
+    #[cfg(gdma)]
+    DmaOutfifoEmptyErr,
+
+    /// Triggered at the end of End_SEG_TRANS transfer in GP-SPI2 slave
+    /// segmented transfer mode or at the end of configurable segmented transfer
+    /// as master.
+    #[cfg(any(esp32s2, gdma))]
+    DmaSegTransDone,
+
+    /// Triggered when a Magic error occurs in CONF buffer during configurable
+    /// segmented transfer as master.
+    #[cfg(any(esp32s2, gdma))]
+    SegMagicErr,
+
+    /// Triggered by RX AFIFO write-full error in GP-SPI2 as master.
+    #[cfg(gdma)]
+    RxAfifoWfullErr,
+
+    /// Triggered by TX AFIFO read-empty error in GP-SPI2 as master.
+    #[cfg(gdma)]
+    TxAfifoRemptyErr,
```

Comment:
Given the current state of the driver, these should never happen? Or at least if it does happen then there's a bug in the driver.

This approach of exposing interrupts might be too simple (for the SPI driver) to stabilize.

Reply:
Interrupt functionality in general won't get stabilized any time soon